parsl/executors/high_throughput/mpi_prefix_composer.py

```diff
@@ -18,13 +18,19 @@ def __str__(self):
         return f"Invalid resource specification options supplied: {self.invalid_keys}"
 
 
-def validate_resource_spec(resource_spec: Dict[str, str]):
+def validate_resource_spec(resource_spec: Dict[str, str], is_mpi_enabled: bool = False):
```

Comment:
I prefer to not specify defaults like `= False` here unless there is a reason (for example, for better usability if the function is exposed to end users). Here I think it's more important to be clear on every use of validate_resource_spec whether it is validating for mpi mode or non-mpi mode, and removing this default will ensure that happens.